src/olympia/amo/management/commands/get_changed_files.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+from datetime import datetime, timedelta
+from os import scandir
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from olympia.addons.models import Addon, Preview
+from olympia.amo.utils import id_to_path
+from olympia.blocklist.utils import datetime_to_ts
+from olympia.files.models import File
+from olympia.git.utils import AddonGitRepository
+from olympia.hero.models import PrimaryHeroImage
+from olympia.users.models import UserProfile
+from olympia.versions.models import Version, VersionPreview
+
+
+def collect_user_pics(since):
+    qs = UserProfile.objects.filter(modified__gt=since).values_list('id', flat=True)
+    return [UserProfile(id=id_).picture_dir for id_ in qs]
```

Comment:
Going one step further... Since we're building new instances... Maybe we should just use `UserProfile....only('id')` instead in the queryset ? And adding some `assertNumQueries()` to make sure we are not doing extra database queries.

On top of this, we should add [`.iterator()`](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#iterator) when evaluating the queryset, to bypass queryset caching and avoid extra memory consumption.

Reply:
https://github.com/mozilla/addons-server/pull/20329/commits/984f94393ce089b2d827451866aeb1ce2627b9cd